src/napari_manual_transforms/_widget.py

```diff
@@ -158,7 +159,27 @@ def _on_layer_event(self, event):
 
     def _update_from_layer(self):
         if self._active is not None:
-            self._model.matrix = self._active.affine.affine_matrix[:3, :3]
+            with self._model.valueChanged.blocked():
+                # Affine matrix decomposition based on
+                # https://math.stackexchange.com/a/1463487
+                affine_matrix = self._active.affine.affine_matrix
+
+                # Extract scaling only from first row of the rotation
+                # matrix as only single scaling factor is allowed as input.
+                scale = np.linalg.norm(affine_matrix[0, :3]).item()
+                self._model.scale = scale
+
+                # Adjust translation based on origin offset.
+                origin = self._model.origin
+                T = np.eye(4)
+                T[:3, -1] = origin
+                self._model.translation = (affine_matrix @ T)[:3, 3] - origin
+
+                # Prevent division by zero by using at least the minimum
+                # scaling value that is allowed as scaling input.
+                self._model.matrix = affine_matrix[:3, :3] / max(scale, MINIMUM_SCALE)
```

Comment:
this `max()` and import of `MINIMUM_SCALE` could be removed if you validate the setting of `_model.scale`

Reply:
Same here, resolved imo.